util/file.py

```diff
@@ -1091,3 +1104,116 @@ def read(self, size):
 
     if outfile is not None:
         outfile.close()
+
+class DBConnection:
+    def __init__(self, db_file=None):
+        if db_file==None:
+            db_file = mkstempfname(suffix='.db')
+        self.conn = sqlite3.connect(db_file, isolation_level='DEFERRED')
+        assert self.conn.isolation_level
+        self.conn.text_factory = sqlite3.OptimizedUnicode
+        self.cur = self.conn.cursor()
+        self.cur.execute("PRAGMA foreign_keys=ON")
+        self.cur.execute("PRAGMA foreign_keys")
+        fk = self.cur.fetchone()
+        log.debug("SQLite version: %s" % sqlite3.sqlite_version)
+        log.debug("SQLite foreign key support: %s" % ((fk and fk[0]) and 'true' or 'false'))
+        self.start()
+    def start(self):
+        pass
+    def __enter__(self):
+        return self
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
```

Comment:
maybe if `__init__` was called with `db_file=None`, we should remember that and then `os.unlink` it upon `__exit__` such that invoking this `with DBConnection() as tmp_db:` would do proper cleanup at the end